### PR TITLE
New bottom bar and relative dimensions for recipe card

### PR DESCRIPTION
**Pull request by GTZL1**

### CODE REVIEW

File: src/mobile/src/main/java/tupperdate/android/home/Home.kt [hunk 1]
```diff
@@ -102,15 +103,18 @@ private fun DisplayRecipeCard(
...
                             onDislike()
+                            Log.d("HOMe", "dislike")
                         }
                         setPos(defaultPos)
                         setDelta(0)
                         //the like() call will change the presentRecipe
                         recipeCard(pos, presentRecipe, boxWidth)
                     } else {
                         recipeCard(pos, presentRecipe, boxWidth)
+                        Log.d("HOMe", "normal")
```
Comment (alexandrepiveteau) [inline src/mobile/src/main/java/tupperdate/android/home/Home.kt]:
```suggestion
```

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,10 @@
+*.iml
+.gradle
+/local.properties
+.idea
+.DS_Store
+/build
+/captures
+.externalNativeBuild
+.cxx
+local.properties
```
Comment (alexandrepiveteau) [inline .gitignore:10]: Please delete this file.

File: src/mobile/src/main/java/tupperdate/android/appbars/mainBottomBar.kt [hunk 3]
```diff
@@ -1,35 +1,54 @@
...
-fun mainBottomBar() {
-    BottomAppBar(backgroundColor = Color.White)
+fun mainBottomBar(
+    onLike: () -> Unit,
+    onDislike: () -> Unit,
+    onReturn: () -> Unit,
+    onRecipeClick: () -> Unit
+) {
+    Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceEvenly)
     {
```
Comment (alexandrepiveteau) [inline src/mobile/src/main/java/tupperdate/android/appbars/mainBottomBar.kt]: Formatting {
}

File: src/mobile/src/main/java/tupperdate/android/appbars/mainBottomBar.kt [hunk 4]
```diff
@@ -1,35 +1,54 @@
...
 
 @Composable
-fun mainBottomBar() {
-    BottomAppBar(backgroundColor = Color.White)
+fun mainBottomBar(
+    onLike: () -> Unit,
+    onDislike: () -> Unit,
+    onReturn: () -> Unit,
+    onRecipeClick: () -> Unit
+) {
```
Comment (alexandrepiveteau) [inline src/mobile/src/main/java/tupperdate/android/appbars/mainBottomBar.kt]: Please use a PascalCase naming scheme (it's the convention for composables), and add a `Modifier` to this composable.

File: src/mobile/src/main/java/tupperdate/android/home/Buttons.kt [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+@Composable
+fun BarButton(
+    size: Int,
+    onClick: () -> Unit,
+    backgroundColor: Color,
+    contentColor: Color,
+    content: @Composable() () -> Unit
+) {
```
Comment (alexandrepiveteau) [inline src/mobile/src/main/java/tupperdate/android/home/Buttons.kt]: Missing a `Modifier`, between the `contentColor` and `content` parameters.

File: src/mobile/src/main/java/tupperdate/android/home/Buttons.kt [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    size: Int,
+    onClick: () -> Unit,
+    backgroundColor: Color,
+    content: @Composable() () -> Unit
+) {
+    BarButton(
+        size = size, onClick = onClick, backgroundColor = backgroundColor,
+        contentColor = Color.White, content
+    )
+}
```
Comment (alexandrepiveteau) [inline src/mobile/src/main/java/tupperdate/android/home/Buttons.kt]: You definitely want to refactor this, for a few reasons :

- You can use default parameters for arguments;
- The scope is a `RowScope`, which is more powerful than no scope at all; and
- Sizes MUST be specified in `Dp`, and SHOULD be passed as a `Modifier` parameter.

```suggestion
@Composable
fun BarButton(
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    foreground: Color = MaterialTheme.colors.primary,
    background: Color = contentColorFor(MaterialTheme.colors.primary),
    content: @Composable RowScope.() -> Unit
) {
   Button(
         modifier = modifier,
         shape = CircleShape,
         onClick = onClick,
         colors = ButtonConstants.defaultButtonColors(
             backgroundColor = background,
             contentColor = foreground
         ),
         // Actually, the prototypes have an elevation. This should be fixed.
         elevation = null
     ) {
         content()
     }
}
```

File: src/mobile/src/main/java/tupperdate/android/home/Buttons.kt [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        ),
+        elevation = null
+    ) {
+        content()
+    }
+}
+
+const val littleButtonSize = 45
+const val normalButtonSize = 50
+const val veryLittleButtonSize = 30
```
Comment (alexandrepiveteau) [inline src/mobile/src/main/java/tupperdate/android/home/Buttons.kt]: Use PascalCase. Maybe put that in a `BarButtonConstants` object.

File: src/mobile/src/main/java/tupperdate/android/home/Home.kt [hunk 8]
```diff
@@ -1,98 +1,146 @@
...
+ */
 fun Home(
-    onButtonClick: () -> Unit,
-    user: AuthenticationApi.Profile?,
-    modifier: Modifier = Modifier,
+    recipeApi: RecipeApi,
+    onChatClick: () -> Unit,
+    onProfileClick: () -> Unit,
+    onReturnClick: () -> Unit,
+    onRecipeClick: () -> Unit
```
Comment (alexandrepiveteau) [inline src/mobile/src/main/java/tupperdate/android/home/Home.kt]: Missing a default `Modifier`.

File: src/mobile/src/main/java/tupperdate/android/home/RecipeCard.kt [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+@Composable
+fun recipeCard(offset: Int, recipe: RecipeApi.Recipe, boxWidth: Dp) {
+    Card(
+        modifier = Modifier.fillMaxSize().offset(x = offset.dp),
+        backgroundColor = (Color.Cyan),
+        shape = RoundedCornerShape(8.dp)
+    ) {
+        Box(Modifier.fillMaxSize())
+        {
```
Comment (alexandrepiveteau) [inline src/mobile/src/main/java/tupperdate/android/home/RecipeCard.kt:37]: Formatting.

File: src/mobile/src/main/java/tupperdate/android/home/RecipeCard.kt [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+import androidx.compose.ui.res.imageResource
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import tupperdate.android.R
+import tupperdate.android.ui.TupperdateTypography
+import tupperdate.api.RecipeApi
+
+@Composable
+fun recipeCard(offset: Int, recipe: RecipeApi.Recipe, boxWidth: Dp) {
```
Comment (alexandrepiveteau) [inline src/mobile/src/main/java/tupperdate/android/home/RecipeCard.kt]: + PascalCase
+ Missing a default `Modifier`

Comment (alexandrepiveteau) [inline src/mobile/src/main/java/tupperdate/android/home/RecipeCard.kt]: Please ALWAYS use a `Dp` when you want to pass a measurement..

File: src/mobile/src/main/java/tupperdate/android/home/RecipeCard.kt [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+    Card(
+        modifier = Modifier.fillMaxSize().offset(x = offset.dp),
+        backgroundColor = (Color.Cyan),
+        shape = RoundedCornerShape(8.dp)
+    ) {
+        Box(Modifier.fillMaxSize())
+        {
+            Image(
+                // TODO modify parameter to use a String url
```
Comment (alexandrepiveteau) [inline src/mobile/src/main/java/tupperdate/android/home/RecipeCard.kt:39]: Not needed for this PR, but you may want to use the https://github.com/chrisbanes/accompanist library to display images in compose.

File: src/mobile/src/main/java/tupperdate/android/home/RecipeCard.kt [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+                imageResource(id = R.drawable.lobster),
+                Modifier.fillMaxWidth(),
+                contentScale = ContentScale.Crop
+            )
+            Row(
+                Modifier.fillMaxWidth().align(Alignment.BottomStart)
+                    .padding(5.dp)
```
Comment (alexandrepiveteau) [inline src/mobile/src/main/java/tupperdate/android/home/RecipeCard.kt]: I don't think we've ever used `5.dp` in the Figma prototypes.

File: src/mobile/src/main/java/tupperdate/android/home/RecipeCard.kt [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+                Column(
+                    Modifier.fillMaxWidth(), horizontalAlignment = Alignment.End,
+                    verticalArrangement = Arrangement.Top
+                ) {
+                    BarButton(veryLittleButtonSize,
+                        onClick = { // TODO fill this onClick
+                        },
+                        Color.Transparent, {
+                            Icon(vectorResource(id = R.drawable.ic_help_outline_white_18dp))
+                        })
```
Comment (alexandrepiveteau) [inline src/mobile/src/main/java/tupperdate/android/home/RecipeCard.kt]: `IconButton` may be a better fit.

File: src/mobile/src/main/java/tupperdate/android/home/RecipeCard.kt [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+                    Text(
+                        recipe.description,
+                        style = TupperdateTypography.body2,
+                        color = Color.White,
+                    )
+                }
+                Column(
+                    Modifier.fillMaxWidth(), horizontalAlignment = Alignment.End,
+                    verticalArrangement = Arrangement.Top
+                ) {
```
Comment (alexandrepiveteau) [inline src/mobile/src/main/java/tupperdate/android/home/RecipeCard.kt]: This column looks like it has a single child. Is it needed ?

Comment (alexandrepiveteau) [inline src/mobile/src/main/java/tupperdate/android/home/RecipeCard.kt]: Please use a single `Row` for all the content, and add a `Spacer` in the center with `Modifier.weight(1f, true)`.

File: src/mobile/src/main/java/tupperdate/android/ui/Colors.kt [hunk 15]
```diff
@@ -20,4 +20,9 @@ val Color.Companion.Flamingo500 get() = Color(0xFFFF99FB)
...
 val Color.Companion.Flamingo700 get() = Color(0xFFCC00C4)
 val Color.Companion.Flamingo800 get() = Color(0xFF990093)
-val Color.Companion.Flamingo900 get() = Color(0xFF660062)
\ No newline at end of file
+val Color.Companion.Flamingo900 get() = Color(0xFF660062)
+
+val Color.Companion.returnButton get() = Color(197, 180, 246)
+val Color.Companion.dislikeButton get() = Color(252, 90, 100)
+val Color.Companion.likeButton get() = Color(142, 200, 255)
+val Color.Companion.recipeAddButton get() = Color(223, 189, 157)
```
Comment (alexandrepiveteau) [inline src/mobile/src/main/java/tupperdate/android/ui/Colors.kt]: Please use PascalCase.

File: src/mobile/src/main/res/drawable/ic_launcher_background.xml [hunk 16]
```diff
@@ -1,170 +1,170 @@
...
-        android:pathData="M69,19L69,89"
-        android:strokeWidth="0.8"
-        android:strokeColor="#33FFFFFF" />
-    <path
-        android:fillColor="#00000000"
-        android:pathData="M79,19L79,89"
-        android:strokeWidth="0.8"
-        android:strokeColor="#33FFFFFF" />
-</vector>
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (alexandrepiveteau) [inline src/mobile/src/main/res/drawable/ic_launcher_background.xml:1]: Is this file needed / relevant ??

File: src/mobile/src/main/java/tupperdate/android/home/Home.kt [hunk 17]
```diff
@@ -1,98 +1,146 @@
...
+                            onDislike()
+                            Log.d("HOMe", "dislike")
+                        }
+                        setPos(defaultPos)
+                        setDelta(0)
+                        //the like() call will change the presentRecipe
+                        recipeCard(pos, presentRecipe, boxWidth)
+                    } else {
+                        recipeCard(pos, presentRecipe, boxWidth)
+                        Log.d("HOMe", "normal")
```
Comment (alexandrepiveteau) [inline src/mobile/src/main/java/tupperdate/android/home/Home.kt]: Do not call callbacks from within composable functions. Use callbacks from the DragListener, for instance, but do NEVER rely on the fact that composable bodies are executed only once (because they are not).

File: src/mobile/src/main/java/tupperdate/android/home/Home.kt [hunk 18]
```diff
@@ -1,98 +1,146 @@
...
+
+                Box(modifier = Modifier.align(Alignment.CenterHorizontally)
+                    .width((boxWidth)).height((boxHeight))
+                    .padding(bottom = (veryLittleButtonSize * 0.8f).dp)
+                    .draggable(Orientation.Horizontal,
+                        onDragStarted = {}, onDrag =
+                        { deltaTemp ->
+                            run {
+                                setPos(pos + (deltaTemp * 0.75).toInt())
+                                setDelta(delta+deltaTemp.toInt())
```
Comment (alexandrepiveteau) [inline src/mobile/src/main/java/tupperdate/android/home/Home.kt]: I would not round this.

File: src/mobile/src/main/java/tupperdate/android/home/Home.kt [hunk 19]
```diff
@@ -1,98 +1,146 @@
...
+                val swipeMargin = (boxWidth).value.toInt()
+
+                Box(modifier = Modifier.align(Alignment.CenterHorizontally)
+                    .width((boxWidth)).height((boxHeight))
+                    .padding(bottom = (veryLittleButtonSize * 0.8f).dp)
+                    .draggable(Orientation.Horizontal,
+                        onDragStarted = {}, onDrag =
+                        { deltaTemp ->
+                            run {
+                                setPos(pos + (deltaTemp * 0.75).toInt())
```
Comment (alexandrepiveteau) [inline src/mobile/src/main/java/tupperdate/android/home/Home.kt]: Why not calculate pos as `defaultPos + delta * 0.75` and use that wherever you need it ? You're duplicating state.

File: src/mobile/src/main/java/tupperdate/android/home/Home.kt [hunk 20]
```diff
@@ -1,98 +1,146 @@
...
+                //the card measures 90 % of screen width and 75 % of screen height
+                val boxHeight =
+                    with(DensityAmbient.current) { constraints.maxHeight.toDp() * 0.75f }
+                val boxWidth = with(DensityAmbient.current) { constraints.maxWidth.toDp() * 0.9f }
+
+                //the align(..) doesn't work anymore (because of reasons), so we need to
+                //set manually a default pos
+                val defaultPos = ((with(DensityAmbient.current) { constraints.maxWidth.toDp() / 2 })
+                        - boxWidth / 2).value.toInt()
+                val (pos, setPos) = remember { mutableStateOf((defaultPos)) }
```
Comment (alexandrepiveteau) [inline src/mobile/src/main/java/tupperdate/android/home/Home.kt]: This is because you're in a `WithConstraints` scope. It might be worth doing things a bit differently; you probably do not need absolute positioning of your composable.

Comment (alexandrepiveteau) [reply src/mobile/src/main/java/tupperdate/android/home/Home.kt]: In particular, the swipe slop does not have to be a percentage of the card width, but probably a fixed size (think `160.dp`). If the user is in landscape (or even worse, on a phablet), he should have to scroll the same amount to dismiss the card, not 10x more because the screen is 10x bigger.

File: src/mobile/src/main/java/tupperdate/android/home/Home.kt [hunk 21]
```diff
@@ -1,98 +1,146 @@
...
+            Modifier.fillMaxWidth().align(Alignment.Bottom)
+                .padding(bottom = veryLittleButtonSize.dp)
+        ) {
+            WithConstraints { -> //why no parameter here ??
+
+                //determine height and width of card according to constraints
+                //the card measures 90 % of screen width and 75 % of screen height
+                val boxHeight =
+                    with(DensityAmbient.current) { constraints.maxHeight.toDp() * 0.75f }
+                val boxWidth = with(DensityAmbient.current) { constraints.maxWidth.toDp() * 0.9f }
```
Comment (alexandrepiveteau) [inline src/mobile/src/main/java/tupperdate/android/home/Home.kt]: The card should fit its content size.

File: src/mobile/src/main/java/tupperdate/android/home/Home.kt [hunk 22]
```diff
@@ -1,98 +1,146 @@
...
-    onButtonClick: () -> Unit,
-    user: AuthenticationApi.Profile,
-    modifier: Modifier,
+private fun DisplayRecipeCard(
+    presentRecipe: RecipeApi.Recipe,
+    onLike: () -> Unit,
+    onDislike: () -> Unit,
+    onReturnClick: () -> Unit,
+    onRecipeClick: () -> Unit
 ) {
```
Comment (alexandrepiveteau) [inline src/mobile/src/main/java/tupperdate/android/home/Home.kt]: I would probably make a `SwipeableStack` composable with a slot `contents: @Composable () -> Unit` that builds children.

Even better, the final signature (with support for stacked items) would look like this :

```
@Composable
fun SwipeableStack<T>(
  items: List<T>,
  modifier: Modifier = Modifier,
  onItemSwiped: ((T) -> Unit)? = {},
  content: @Composable (T) -> Unit,
)
```

File: src/mobile/src/main/java/tupperdate/android/appbars/mainTopBar.kt [hunk 23]
```diff
@@ -1,27 +1,55 @@
...
+import tupperdate.android.home.normalButtonSize
+import tupperdate.android.ui.TupperdateTypography
 
 @Composable
-fun mainTopBar() {
-    TopAppBar(
-        title = {
-            Column(modifier = Modifier.weight(1f), horizontalGravity = Alignment.CenterHorizontally) {
-                Text(text = "tupper.date", textAlign = TextAlign.Center)
+fun mainTopBar(
```
Comment (alexandrepiveteau) [inline src/mobile/src/main/java/tupperdate/android/appbars/mainTopBar.kt]: PascalCase.

File: src/mobile/src/main/java/tupperdate/android/appbars/mainTopBar.kt [hunk 24]
```diff
@@ -1,27 +1,55 @@
...
+    onChatClick: () -> Unit,
+    onProfileClick: () -> Unit
+) {
```
Comment (alexandrepiveteau) [inline src/mobile/src/main/java/tupperdate/android/appbars/mainTopBar.kt]: Missing a default Modifier.

File: src/mobile/src/main/java/tupperdate/android/appbars/mainTopBar.kt [hunk 25]
```diff
@@ -1,27 +1,55 @@
...
+    TopAppBar(backgroundColor = Color.Transparent, elevation = 0.dp) {
+        Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween)
+        {
```
Comment (alexandrepiveteau) [inline src/mobile/src/main/java/tupperdate/android/appbars/mainTopBar.kt]: Formatting.

**Review by alexandrepiveteau — CHANGES_REQUESTED** [review]

File: src/mobile/src/main/java/tupperdate/android/appbars/mainTopBar.kt (hunk 24 above)
Comment (GTZL1) [reply src/mobile/src/main/java/tupperdate/android/appbars/mainTopBar.kt]: We don't need it. G-L told me to remove them in this case

File: src/mobile/src/main/java/tupperdate/android/home/Home.kt (hunk 21 above)
Comment (GTZL1) [reply src/mobile/src/main/java/tupperdate/android/home/Home.kt]: maxWidth is the size of the screen. We don't want the card to occupy it entierly.

File: src/mobile/src/main/java/tupperdate/android/home/Buttons.kt (hunk 5 above)
Comment (GTZL1) [reply src/mobile/src/main/java/tupperdate/android/home/Buttons.kt]: We don't need it. G-L told me to remove them in this case.

File: src/mobile/src/main/java/tupperdate/android/home/Home.kt (hunk 1 above)
Comment (GTZL1) [reply src/mobile/src/main/java/tupperdate/android/home/Home.kt]: I left them for you to test it

File: src/mobile/src/main/java/tupperdate/android/home/Buttons.kt (hunk 5 above)
Comment (glsubri) [reply src/mobile/src/main/java/tupperdate/android/home/Buttons.kt]: What ?

File: src/mobile/src/main/java/tupperdate/android/appbars/MainBottomBar.kt [hunk 26]
```diff
@@ -0,0 +1,55 @@
...
+import tupperdate.android.home.BarButton
+import tupperdate.android.home.littleButtonSize
+import tupperdate.android.home.normalButtonSize
+import tupperdate.android.ui.dislikeButton
+import tupperdate.android.ui.likeButton
+import tupperdate.android.ui.recipeAddButton
+import tupperdate.android.ui.returnButton
+
+@Composable
+fun MainBottomBar(
```
Comment (glsubri) [inline src/mobile/src/main/java/tupperdate/android/appbars/MainBottomBar.kt]: I think we could find a better name than bottom bar, especially since it's closely related to the recipe card.
Also, we could move that file to the home package

File: src/mobile/src/main/java/tupperdate/android/appbars/TitleTopBar.kt [hunk 27]
```diff
@@ -0,0 +1,63 @@
...
+            BarButton(
+                normalButtonSize,
+                { onChatClick() },
+                Color.Transparent,
+                Color.Black,
+                { Icon(vectorResource(id = R.drawable.ic_chat_24px)) }
+            )
+            Column(
+                modifier = Modifier.align(Alignment.CenterVertically)
+            ) {
```
Comment (glsubri) [inline src/mobile/src/main/java/tupperdate/android/appbars/TitleTopBar.kt:46]: Pretty sure we could remove that Column

File: src/mobile/src/main/java/tupperdate/android/home/Buttons.kt [hunk 28]
```diff
@@ -0,0 +1,52 @@
...
+    onClick: () -> Unit,
+    backgroundColor: Color,
+    content: @Composable() () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    BarButton(
+        modifier = modifier, size = size,
+        onClick = onClick, backgroundColor = backgroundColor,
+        contentColor = Color.White, content = content
+    )
```
Comment (glsubri) [inline src/mobile/src/main/java/tupperdate/android/home/Buttons.kt]: formatting

File: src/mobile/src/main/java/tupperdate/android/home/Buttons.kt [hunk 29]
```diff
@@ -0,0 +1,52 @@
...
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun BarButton(
+    size: Int,
+    onClick: () -> Unit,
+    backgroundColor: Color,
+    content: @Composable() () -> Unit,
```
Comment (glsubri) [inline src/mobile/src/main/java/tupperdate/android/home/Buttons.kt]: I'm not sure, but shouldn't the content be the last element to be able to use Kotlin's trailing lambda ?

File: src/mobile/src/main/java/tupperdate/android/home/Buttons.kt [hunk 30]
```diff
@@ -0,0 +1,52 @@
...
+    BarButton(
+        modifier = modifier, size = size,
+        onClick = onClick, backgroundColor = backgroundColor,
+        contentColor = Color.White, content = content
+    )
+}
+
+@Composable
+fun BarButton(
+    size: Int,
```
Comment (glsubri) [inline src/mobile/src/main/java/tupperdate/android/home/Buttons.kt]: Shouldn't the `size` be in dp ?

File: src/mobile/src/main/java/tupperdate/android/home/Home.kt [hunk 31]
```diff
@@ -1,98 +1,145 @@
...
         }
     }
 }
 
 @Composable
-private fun HomeConnected(
-    onButtonClick: () -> Unit,
-    user: AuthenticationApi.Profile,
-    modifier: Modifier,
+private fun DisplayRecipeCard(
```
Comment (glsubri) [inline src/mobile/src/main/java/tupperdate/android/home/Home.kt:62]: The height of the picture doesn't seem to be correct, but we can fix that later. If so, we should add an issue

File: src/mobile/src/main/java/tupperdate/android/home/RecipeCard.kt (hunk 14 above)
Comment (GTZL1) [reply src/mobile/src/main/java/tupperdate/android/home/RecipeCard.kt]: The button one no :)

Comment (GTZL1) [reply src/mobile/src/main/java/tupperdate/android/home/RecipeCard.kt]: The column within the row is needed to display the texts one above other

File: src/mobile/src/main/res/drawable/ic_launcher_background.xml (hunk 16 above)
Comment (GTZL1) [reply src/mobile/src/main/res/drawable/ic_launcher_background.xml:1]: I don't know, I didn't create it

File: src/mobile/src/main/java/tupperdate/android/ui/Colors.kt (hunk 15 above)
Comment (GTZL1) [reply src/mobile/src/main/java/tupperdate/android/ui/Colors.kt]: Ok

File: src/mobile/src/main/java/tupperdate/android/home/RecipeCard.kt (hunk 10 above)
Comment (GTZL1) [reply src/mobile/src/main/java/tupperdate/android/home/RecipeCard.kt]: here we can't, pos is an int mutable state

File: src/mobile/src/main/java/tupperdate/android/home/RecipeCard.kt (hunk 13 above)
Comment (GTZL1) [reply src/mobile/src/main/java/tupperdate/android/home/RecipeCard.kt]: No, we're already in a Button, this icon is its content

File: src/mobile/src/main/java/tupperdate/android/appbars/TitleTopBar.kt (hunk 27 above)
Comment (GTZL1) [reply src/mobile/src/main/java/tupperdate/android/appbars/TitleTopBar.kt:46]: It is needed for the CenterVertically

File: src/mobile/src/main/java/tupperdate/android/home/Buttons.kt (hunk 28 above)
Comment (GTZL1) [reply src/mobile/src/main/java/tupperdate/android/home/Buttons.kt]: It has been changed already

File: src/mobile/src/main/java/tupperdate/android/home/Buttons.kt (hunk 29 above)
Comment (GTZL1) [reply src/mobile/src/main/java/tupperdate/android/home/Buttons.kt]: I don't know, but this is working fine

File: src/mobile/src/main/java/tupperdate/android/home/Buttons.kt (hunk 30 above)
Comment (GTZL1) [reply src/mobile/src/main/java/tupperdate/android/home/Buttons.kt]: Ok

File: src/mobile/src/main/java/tupperdate/android/home/Home.kt (hunk 31 above)
Comment (GTZL1) [reply src/mobile/src/main/java/tupperdate/android/home/Home.kt:62]: I just tested, and it changes with 0.75 or 0.8 parameter. So it works !

File: src/mobile/src/main/java/tupperdate/android/appbars/MainBottomBar.kt (hunk 26 above)
Comment (GTZL1) [reply src/mobile/src/main/java/tupperdate/android/appbars/MainBottomBar.kt]: Ok
